File: C/SEM I/Practical/Done/27.c
/* Write a program to find the sum of the following series:
1! + 2! + 3! + 4! + 5! ... up to n terms */

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>
#include <math.h>

int count = 0;

int main(void) {
	int rep = 1;

	while (rep == 1) {
		float sum = 0, term;
		int n;
		int i, j;

		printf("Enter the number of terms upto which sum is to be calculated: ");
		scanf("%d", & n);

		for (i = 1; i <= n; i++) {
			term = 1;

			for (j = 1; j <= i; j++) {
				term *= j;
			}

			sum += term;

			printf("%.0f", term);

			if (i != n) {
				printf(" + ");
			}
		}

		printf("\n= %.0f", sum);

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 50) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		}

	}

	printf("Normal END\n");

	getch();
	return 0;
}

File: Python/practical/27.py
# Program to find sum of factorials

import math

def sumFind(num):
    if num <= 0: return 0
    
    termSum = 0
    
    for i in range(1, num + 1):
        termSum = termSum + math.factorial(i)
        
        print(f"{math.factorial(i)}", end = '')
        
        if i != num:
            print (' + ', end = '')
            
    return termSum

num = int(input("Enter the number of terms: "))

termSum = sumFind(num)

print(f' = {termSum}')